BookPlayer/Utils/Extensions/View+BookPlayer.swift

```diff
@@ -36,3 +37,39 @@ extension View {
       .opacity(condition ? opacity : 1)
   }
 }
+
+struct MiniPlayerSafeAreaInsetModifier: ViewModifier {
+  let spacing: CGFloat
+
+  @Environment(\.playerState) var playerState
+
+  func body(content: Content) -> some View {
+    content
+      .safeAreaInset(edge: .bottom) {
+        Spacer().frame(
+          height: playerState.loadedBookRelativePath != nil
+            ? spacing
+            : Spacing.M
+        )
+      }
+  }
+}
+
+extension View {
```

Comment:
❓ Can `MainView` directly handle the safe area adjustment when the mini player is visible?

Reply:
I tried, but it doesn't quite work with how the safe area inset is already being used for the mini player, I can use a ZStack to overlay it, but I would still need to add the padding inside each tab root view